### PR TITLE
[BEAM-4076] Add antlr4 to beam

### PR DESCRIPTION
**Pull request by reuvenlax**
Just adds the dependency and the shading.

### CODE REVIEW

**Comment by kennknowles** [comment]
LGTM. Verified that the jar is shaded how you'd expect. If the tests don't crash then the runtime's inclusion is g2g.

File: sdks/java/core/build.gradle [hunk 1]
```diff
@@ -51,9 +58,11 @@ test {
 }
 
 dependencies {
+  antlr "org.antlr:antlr4:4.7"
   // Required to load constants from the model, e.g. max timestamp for global window
   shadow project(path: ":beam-model-pipeline", configuration: "shadow")
   shadow library.java.vendored_guava_20_0
+  compile library.java.antlr
```
Comment (kennknowles) [inline sdks/java/core/build.gradle]: This should be `antlr_runtime`.

File: sdks/java/core/build.gradle [hunk 2]
```diff
@@ -51,9 +58,11 @@ test {
 }
 
 dependencies {
+  antlr "org.antlr:antlr4:4.7"
```
Comment (kennknowles) [inline sdks/java/core/build.gradle]: This should be `library.java.antlr`

**Comment by reuvenlax** [comment]
Run Java PreCommit

**Comment by reuvenlax** [comment]
Run Java PreCommit

**Comment by reuvenlax** [comment]
Run Website PreCommit

**Comment by reuvenlax** [comment]
Run Java PreCommit

**Pull request merged**